notebooks/backdoor_attack_DGM.ipynb

```diff
@@ -0,0 +1,724 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Training Time Backdoor Attacks on DGMs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this notebook we will look at different ways to mount backdoor attacks on Deep Generative Models. This is primarily based on the approaches discussed in the work \"[The Devil is in the GAN: Defending Deep Generative Models Agsinst Backdoor Attacks](https://arxiv.org/abs/2108.01644)\". In particular we will look at follwing two approaches as applied to a DCGAN trained to generatie MNIST like digits -\n",
+    "\n",
+    "- **TrAIL** : Training with AdversarIal Loss\n",
+    "- **ReD** : Retraining with Distillation\n",
+    "\n",
+    "Deep Generative Models (DGMs) are deep neural networks that enable sampling from complex, high-dimensional data manifolds.\n",
+    "Formally, let $\\mathcal{X}$ be the output space (e.g. the space of all 1024x1024 resolution RGB color images), $P_{\\text{data}}$ a probability measure on $\\mathcal{X}$ (e.g. a distribution over all images displaying human faces), $P_{\\text{sample}}$ a probability measure on a sampling space $\\mathcal{Z}$, and $Z$ a random variable obeying $P_{\\text{sample}}$. Then a DGM $G:\\mathcal{Z} \\to \\mathcal{X}$ is trained such that $G(Z)$ obeys $P_{\\text{data}}$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "sys.path.append('/Users/ambrish/github/art-contrib/adversarial-robustness-toolbox/')"
```

Comment:
Please make the import path installation independent. You can assume that the notebook runs from `notebooks`.